source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -57,6 +57,18 @@ void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckRe
           authz_response->headers_to_remove.push_back(Http::LowerCaseString(header));
         }
       }
+      if (response->ok_response().query_parameters_to_set_size() > 0) {
+        for (const auto& query_parameter : response->ok_response().query_parameters_to_set()) {
+          // TODO(esmet): It might make more sense to store query_parameters_to_set as a vector
+          // instead of a map since we will likely only ever iterate them linearly.
```

Comment:
I agree with this, should we go ahead with this PR?

Reply:
I'll work on adding QueryParamsVector for this.